src/components/menu/menu.types.ts

```diff
@@ -1,4 +1,14 @@
-export type OpenDirection = 'left' | 'right';
+export type OpenDirection =
+    | 'left'
+    | 'right'
+    | 'bottom-center'
+    | 'top-center'
+    | 'left-bottom'
+    | 'left-center'
+    | 'left-top'
+    | 'right-bottom'
+    | 'right-center'
+    | 'right-top';
```

Comment:
Great job. A few notes on the namings:

### 1. Naming 
I have some problems with these naming conventions. Maybe that's only me... what do you think?
I think what the popperjs does is quite OK. They simply use `start´ and `end` prefixes. 

Example:
| consistent naming       | visual naming     |  |
|--------------|-----------|------------|
| left-start | let-top      |        |
| left      | left-center  |       |
| left-end      | left-bottom |       |
| top-start | top-left      |        |
| top      | top-center  |       |
| top-end      | top-left  |       |

an so on…

### 2. Missing locations
`top` and `bottom` do not have `left` and `right` locations.

### 3. Duplicates
`left` and `left-center` are the same 
`right` and `right-center` too :)

Reply:
> ### 1. Naming
> I have some problems with these naming conventions. Maybe that's only me... what do you think? I think what the popperjs does is quite OK. They simply use `start` and `end` prefixes. Not a big deal if we don't change them.
> 
> Example:
> 
> consistent naming	visual naming	
> left-start	let-top	
> left	left-center	
> left-end	left-bottom	
> top-start	top-left	
> top	top-center	
> top-end	top-left	
> an so on…

I don't quite follow. Could you be more explicit with what it is you think should be changed, and to what? I don't understand what "They simply use `start` and `end` prefixes" means 🤔

Reply:
> They simply use `start` and `end` prefixes.

Thanks, I see your point. Maybe you are right, I just want to explain why I named it with the left and right instead. We are all different and someone `starts` from the left - so `it's LEFT` others from the `right` and `it's their start point` like reading a sentence. I would say it could be even okay with the horizontal direction - as most of the users are in Europe and probably they get used to `starting as a left` and `end as the right`. BUT vertical is a bit confusing since I `wouldn’t say the top is a start` usually the `bottom is when we start`. but again it's the opposite for the reading. 🤔  I hope I didn't confuse you more 😁

Reply:
> ### 2. Missing locations
> `top` and `bottom` do not have `left` and `right` locations.

We use it for the `right` and `left`
   ```
        let placement: Placement = 'bottom-start';   // default right position
        let flipPlacement: Placement = 'top-start';

        if (this.openDirection === 'left') {.  // left position
            placement = 'bottom-end';
            flipPlacement = 'top-end';
        }
```

Reply:
> ### 3. Duplicates
> `left` and `left-center` are the same `right` and `right-center` too :)

You are right, but, Unfortunately again because of the previous naming we will lose left center placement and right center placement since they are taken for the : 
     ` let placement: Placement = 'bottom-start';
        let flipPlacement: Placement = 'top-start';

        if (this.openDirection === 'left') {
            placement = 'bottom-end';
            flipPlacement = 'top-end';
        }`

But we wanted to have a center position as an option SO to distinguish them a bit I named `left-center` and `right-center`

Reply:
I think it's a good idea to stick with what popper is using, if it's not too late already.

I think that `left` should be used if the it stays left regardless of whether the chosen locale is `ltr` or `rtl`. (And conversely for `right` of course.) But if the menu is on the left side when Swedish is selected, and moves to the right side when Arabic is selected (if we supported Arabic), then `left` is a very bad name to describe that behaviour. Then `start` is much better (and a widely used convention nowadays).

> I wouldn’t say the top is a start usually the bottom is when we start

😮 🤨 Maybe we're thinking of different examples, but I definitely associate the _top_ to be the starting point, not the _bottom_. Unless building a house, sure. But for any examples I can think of that feels relevant to this context. What examples are you thinking of? 🙂

Reply:
> I don't quite understand what's going on here. But that sounds like a mistake in the code, and something we should fix regardless. Because `bottom-end` and `top-end` are definitely located on the right side of the element in popper, not on the left side. 🤔 Or?

left has 
    `      if (this.openDirection === 'left') {
            placement = 'bottom-end';
            flipPlacement = 'top-end';
        }`

and it looks: <img width="241" alt="Screenshot 2022-07-06 at 15 47 26" src="https://user-images.githubusercontent.com/50618208/177565293-6eed31fd-a54a-4b64-bdcc-6ce29e593635.png">

So it's up to us keep or change - but it's how it was before

Reply:
🤔 Not sure why that decision was made, and why they call it `left`. Maybe they thought this would mostly be used for menus, which are typically located on the top-right side of the screen? 

I have no idea. 🤷‍♂️ But I feel we have to fix this, and default the location either to something else. Or not call that default thing `left`, and instead correctly call it `bottom-end`. 

I'm super confused, and don't know if I'm thinking straight now. 

Changing the default opening directions may become a breaking change, but not a super horrible one. Because the portal will still do its best to place things in the viewport as I've understood it. However, that would mean we need to adjust the open direction in a few of our components, like the `limec-property-filter`. 

Does anyone have a good suggestion? @adrianschmidt @jgroth ?

Reply:
@eketorp Not because we need your input, but because you should read the thread from the start up to this point. It might give you some answers as to why so often want to take some time to consider the bigger picture even if the task at hand is just to create a minimum viable version of something 😄 (I'm _not_ saying this was brought about by a Product Owner, not at all. But it's a great example of how even trivial things can cause significant headache further down the road.)

Reply:
I just updated the naming in the correct way.🤓🙈😁 Let's talk about it tomorrow. (I will commit after it)

Reply:
> The default placement has been called `left`

No, 😀 the default one is the 'right' direction
![Screenshot 2022-07-06 at 18 54 11](https://user-images.githubusercontent.com/50618208/177603344-8c8b4293-e58f-4908-98e3-269a5700ff93.png)

BUT you are right we can say **`bottom-start`** is a default one and we will get the same UI:
<img width="191" alt="Screenshot 2022-07-06 at 18 55 28" src="https://user-images.githubusercontent.com/50618208/177603681-a52aeb55-3441-49fa-a22d-4721bd29e713.png">

Reply:
The problem will be only if the `left` position was specified. in this case they will get:
<img width="308" alt="Screenshot 2022-07-07 at 07 48 43" src="https://user-images.githubusercontent.com/50618208/177729647-f0d27fb4-0c1a-4900-86dd-2fa2e01ba6f0.png">
 
But expected behaviour is: 
<img width="178" alt="Screenshot 2022-07-07 at 10 36 52" src="https://user-images.githubusercontent.com/50618208/177730079-df9fe71d-60a1-4a18-88e6-dbd59aef3882.png">

Reply:
> The problem will be only if the `left` position was specified.

Or if someone explicitly specified `right`. So we should mark the change as breaking, but with Kia's excellent research (❤️), we can be confident this change will not cause us any significant problem 👍